programs/token/src/state/token.rs

```diff
@@ -89,16 +89,19 @@ impl TokenAccount {
         if account_info.owner() != &ID {
             return Err(ProgramError::InvalidAccountData);
         }
-        Ok(Self::from_bytes(account_info.borrow_data_unchecked()))
+        Ok(Self::from_bytes_unchecked(
+            account_info.borrow_data_unchecked(),
+        ))
     }
 
     /// Return a `TokenAccount` from the given bytes.
     ///
     /// # Safety
     ///
-    /// The caller must ensure that `bytes` contains a valid representation of `TokenAccount`.
+    /// The caller must ensure that `bytes` contains a valid representation of `TokenAccount` and its aligned to 1 byte. This method does not
+    /// perform a length validation.
```

Comment:
```suggestion
    /// The caller must ensure that `bytes` contains a valid representation of `TokenAccount`, and
    /// it is properly aligned to be interpreted as an instance of `TokenAccount`.
    /// At the moment `TokenAccount` has an alignment of 1 byte.
    /// This method does not perform a length validation.
```

We format code and comments not to exceed 100 characters in with, I think.

Also, the alignment requirement of 1 byte is somewhat confusing.
It essentially says "any alignment works".
It would be more accurate so say that this method is only safe when `bytes` is aligned correctly for the current alignment of `TokenAccount`.
Which is 1 for the moment.